homeassistant/components/renson/binary_sensor.py

```diff
@@ -0,0 +1,133 @@
+"""Binary sensors for renson."""
+from dataclasses import dataclass
+
+from renson_endura_delta.field_enum import (
+    AIR_QUALITY_CONTROL_FIELD,
+    BREEZE_ENABLE_FIELD,
+    BREEZE_MET_FIELD,
+    CO2_CONTROL_FIELD,
+    FROST_PROTECTION_FIELD,
+    HUMIDITY_CONTROL_FIELD,
+    PREHEATER_FIELD,
+    DataType,
+    FieldEnum,
+)
+from renson_endura_delta.renson import RensonVentilation
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import RensonCoordinator
+from .const import DOMAIN
+from .entity import RensonEntity
+
+
+@dataclass
+class RensonBinarySensorEntityDescriptionMixin:
+    """Mixin for required keys."""
+
+    field: FieldEnum
+
+
+@dataclass
+class RensonBinarySensorEntityDescription(
+    BinarySensorEntityDescription, RensonBinarySensorEntityDescriptionMixin
+):
+    """Description of binary sensor."""
+
+
+BINARY_SENSORS: tuple[RensonBinarySensorEntityDescription, ...] = (
+    RensonBinarySensorEntityDescription(
+        name="Frost protection active",
+        key="FROST_PROTECTION_FIELD",
+        field=FROST_PROTECTION_FIELD,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="BREEZE_ENABLE_FIELD",
+        name="Breeze",
+        field=BREEZE_ENABLE_FIELD,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="BREEZE_MET_FIELD",
+        name="Breeze conditions met",
+        field=BREEZE_MET_FIELD,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="HUMIDITY_CONTROL_FIELD",
+        name="Humidity control",
+        field=HUMIDITY_CONTROL_FIELD,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="AIR_QUALITY_CONTROL_FIELD",
+        name="Air quality control",
+        field=AIR_QUALITY_CONTROL_FIELD,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="CO2_CONTROL_FIELD",
+        name="CO2 control",
+        field=CO2_CONTROL_FIELD,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+    RensonBinarySensorEntityDescription(
+        key="PREHEATER_FIELD",
+        name="Preheater",
+        field=PREHEATER_FIELD,
+        entity_category=EntityCategory.DIAGNOSTIC,
+    ),
+)
+
+
+class RensonBinarySensor(RensonEntity, BinarySensorEntity):
```

Comment:
It seems `RensonEntity` does not set the `has_entity_name` flag.
`RensonEntity` hence needs to be updated according to this: https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations